elixir/apps/web/lib/web/controllers/auth_controller.ex

```diff
@@ -70,15 +84,31 @@ defmodule Web.AuthController do
           }
         } = params
       ) do
-    _ =
+    conn =
       with {:ok, provider} <- Domain.Auth.fetch_active_provider_by_id(provider_id),
            {:ok, identity} <-
-             Domain.Auth.fetch_identity_by_provider_and_identifier(provider, provider_identifier),
+             Domain.Auth.fetch_identity_by_provider_and_identifier(provider, provider_identifier,
+               preload: :account
+             ),
            {:ok, identity} <- Domain.Auth.Adapters.Email.request_sign_in_token(identity) do
         sign_in_link_params = Map.take(params, ["client_platform", "client_csrf_token"])
 
-        Web.Mailer.AuthEmail.sign_in_link_email(identity, sign_in_link_params)
-        |> Web.Mailer.deliver()
+        <<email_secret::binary-size(5), browser_secret::binary>> =
+          identity.provider_virtual_state.sign_in_token
+
+        {:ok, _} =
+          Web.Mailer.AuthEmail.sign_in_link_email(
+            identity,
+            email_secret,
+            conn.assigns.user_agent,
+            conn.remote_ip,
+            sign_in_link_params
+          )
+          |> Web.Mailer.deliver()
+
+        put_session(conn, :browser_csrf_token, browser_secret)
```

Comment:
What would you think of calling this something like `:sign_in_nonce`?

I don't believe this is the case, but if this is actually a CSRF token, we need to make sure we're not putting it in the cookie.

Reply:
👍 for `sign_in_nonce`. Pushed a fix.